### DISCUSSION TITLE
Legal Moves for a single cell

### DISCUSSION THREAD

**Discussion by ShebinJoseJacob** [niklasf/python-chess#739] [Q&A] [answered]
I saw the function to generate all legal moves for the entire board. Is there any function that can generate legal moves for a single cell? Any Ideas to do this task?

**Comment by niklasf** [comment, marked as answer]
You can filter legal moves by masks like

```
>>> list(board.generate_legal_moves(from_mask=chess.BB_B1))
[Move.from_uci('b1c3'), Move.from_uci('b1a3')]
```

Other useful masks: https://python-chess.readthedocs.io/en/latest/core.html#chess.BB_SQUARES

**Reply by ShebinJoseJacob** [reply]
Thanks. It was really quick.
